am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -122,4 +126,41 @@ public FilterResourceResponse filterResource(String userId, String resourceId, J
 
         return null;
     }
+
+
+    /**
+     * Returns a map of JsonPointers and set of permissions when record with user role and type exists in database.
+     *
+     * @param serviceName  (service_name)
+     * @param resourceType (resource_type)
+     * @param resourceName (resource_name)
+     * @param roleNames    A list of (role_names)
+     * @return permissionsByTypeAndRole or null
+     */
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(
+        String serviceName, String resourceType, String resourceName, @NonNull List<String> roleNames) {
```

Comment:
Out of curiosity why did you decide to keep list despite my suggestion to change it?

Reply:
Why it is a list not a set? Is accepting duplicates wise?

Reply:
Why @NotNull is only one last arg not all of them?

Reply:
@geek-soft You said you would like to discuss with @yarpo. Happy to change it knowing the outcome.

Reply:
👍

Reply:
changed to set